src/xmrEngine.js

```diff
@@ -251,23 +251,19 @@ export class MoneroEngine {
         this.walletLocalData.transactionsObj[PRIMARY_CURRENCY]
       const edgeTx = transactionsArray[idx]
 
-      if (edgeTransaction.blockHeight) {
-        // Only update old transactions if the incoming tx is confirmed
-        // Unconfirmed txs will sometimes have incorrect values
-        if (
-          edgeTx.blockHeight !== edgeTransaction.blockHeight ||
-          edgeTx.networkFee !== edgeTransaction.networkFee ||
-          edgeTx.nativeAmount !== edgeTransaction.nativeAmount
-        ) {
-          this.log(`Update transaction: ${tx.hash} height:${tx.blockNumber}`)
-          this.updateTransaction(PRIMARY_CURRENCY, edgeTransaction, idx)
-          this.edgeTxLibCallbacks.onTransactionsChanged(
-            this.transactionsChangedArray
-          )
-          this.transactionsChangedArray = []
-        } else {
-          // this.log(sprintf('Old transaction. No Update: %s', tx.hash))
+      if (edgeTx.blockHeight !== edgeTransaction.blockHeight) {
```

Comment:
The old code seems like it should have solved the issue. You would only update a tx in the db if the new one was confirmed. Even the comment implied the old code was aware of inaccurate amounts prior to a confirmation. I'll merge for now, but let's discuss afterwards.